coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -2037,7 +2037,18 @@ func InsertWorkspaceResource(ctx context.Context, db database.Store, jobID uuid.
 				Health:               health,
 				DisplayOrder:         int32(app.Order),
 				Hidden:               app.Hidden,
-				OpenIn:               app.OpenIn,
+				OpenIn: func() database.WorkspaceAppOpenIn {
+					switch app.OpenIn {
+					case sdkproto.AppOpenIn_WINDOW:
+						return database.WorkspaceAppOpenInWindow
+					case sdkproto.AppOpenIn_SLIM_WINDOW:
+						return database.WorkspaceAppOpenInSlimWindow
+					case sdkproto.AppOpenIn_TAB:
+						return database.WorkspaceAppOpenInTab
+					default:
+						return database.WorkspaceAppOpenInSlimWindow
```

Comment:
curious, is `default` required here?

Reply:
If you look at `provisioner/terraform/resources.go`, you will notice that we handle missing properties (backward compatibility) there:

```

			// Default to the resource name if none is set!
			if attrs.Slug == "" {
				attrs.Slug = resource.Name
			}
			if attrs.DisplayName == "" {
				if attrs.Name != "" {
					// Name is deprecated but still accepted.
					attrs.DisplayName = attrs.Name
				} else {
					attrs.DisplayName = attrs.Slug
				}
			}
```

We may want to move the default logic for `database.WorkspaceAppOpenInSlimWindow` to `resources.go`.